homework5_2+.py

```diff
@@ -44,71 +28,75 @@
     ('PL', ['67 ', '70 ', '72 ', '75 ', '76 ', '80 ', '82 ', '84 ', '87 ']),
     ('PT', ['58 ', '61 ', '62 ', '65 ', '70 ', '74 ', '77 ', '79 ', '81 ']),
-    ('RO', ['47 ', '54 ', '58 ', '61 b', '68 ', '72 ', '76 ', '81 ', '84 ']),
+    ('RO', ['47 ', '54 ', '58 ', '61 ', '68 ', '72 ', '76 ', '81 ', '84 ']),
     ('RS', [': ', ': ', ': ', ': ', '64 ', ': ', '68 ', '73 ', '80 ']),
-    ('SE', ['91 ', '92 ', '93 ', '90 ', '91 ', '94 b', '95 ', '93 ', '96 ']),
+    ('SE', ['91 ', '92 ', '93 ', '90 ', '91 ', '94', '95 ', '93 ', '96 ']),
     ('SI', ['73 ', '74 ', '76 ', '77 ', '78 ', '78 ', '82 ', '87 ', '89 ']),
     ('SK', ['71 ', '75 ', '78 ', '78 ', '79 ', '81 ', '81 ', '81 ', '82 ']),
     ('TR', [': ', '47 ', '49 ', '60 ', '70 ', '76 ', '81 ', '84 ', '88 ']),
     ('UK', ['83 ', '87 ', '88 ', '90 ', '91 ', '93 ', '94 ', '95 ', '96 ']),
     ('XK', [': ', ': ', ': ', ': ', ': ', ': ', '89 ', '93 ', '93 ']),
 ]
 
-dataset = {'Romania': [{'year': '2019', 'coverage': 84},
-                       {'year': '2018', 'coverage': 81},
-                       {'year': '2017', 'coverage': 76},
-                       {'year': '2016', 'coverage': 72},
-                       {'year': '2015', 'coverage': 68},
-                       {'year': '2014', 'coverage': 61},
-                       {'year': '2013', 'coverage': 58},
-                       {'year': '2012', 'coverage': 54},
-                       {'year': '2011', 'coverage': 47},
-                       ], 'Germany': [{'year': '2019', 'coverage': 95},
-                                      {'year': '2018', 'coverage': 94},
-                                      {'year': '2017', 'coverage': 93},
-                                      {'year': '2016', 'coverage': 92},
-                                      {'year': '2015', 'coverage': 90},
-                                      {'year': '2014', 'coverage': 89},
-                                      {'year': '2013', 'coverage': 88},
-                                      {'year': '2012', 'coverage': 85},
-                                      {'year': '2011', 'coverage': 83},
-                                      ]}
 
-print(dataset.keys())
-print(dataset.values())
-print('============================================================')
+def process_data(data_to_process):
+    years = description[1]
+    dataset_data_to_process = {
+        country: [
+            {'year': year.strip(), 'coverage': coverage.strip() if coverage != ": " else "0"}
+            for coverage, year in zip(coverages, years)
+        ]
+        for country, coverages in data_to_process
+    }
+
+    pprint(dataset_data_to_process)
+    return dataset_data_to_process
 
 
-# get_country_data(dataset,country)
-def get_country_data(dataset, country):
-    for i in range(len(dataset[(f"{country}")])):
+def get_country_data(dataset_get_country_data, country):
+    for i in range(len(dataset_get_country_data[f"{country}"])):
```

Comment:
country is already a string in this case so is not necessary to convert it. Usually for converting to string we use `str()`

Try to iterate without an index